mypy/stubtest.py

```diff
@@ -1058,7 +1072,9 @@ def test_stubs(args: argparse.Namespace) -> int:
 
     # Print unused whitelist entries
     for w in whitelist:
-        if not whitelist[w]:
+        # Don't consider an entry unused if it regex-matches the empty string
+        # This allows us to whitelist errors that don't manifest at all on some systems
+        if not whitelist[w] and not whitelist_regexes[w].fullmatch(""):
```

Comment:
I'm not sure I understand what's going on here. Could you give an example of a situation where this would apply?

Reply:
The motivation here is to create a way to provide a better experience for cases like https://github.com/python/typeshed/issues/3782 where the whitelisted errors don't match up on some user's system.

For instance, on my system running stubtest gives me a number of (unfixable within current type system) errors like `termios.B921600 is not present at runtime`, but in CI we don't have any errors from `termios` at all. If we added a whitelist entry for `termios`, it would complain about an unused whitelist entry in CI. This allows us to add an entry for `(termios.[A-Z0-9_]+)?` so things work both on CI and on user's systems.

This is tested by the `unused_missing` part of `test_whitelist`